clusters/build-clusters/01_cluster/loki.yaml

```diff
@@ -0,0 +1,349 @@
+        - __service__
+        target_label: job
+      - action: replace
+        source_labels:
+        - __meta_kubernetes_namespace
+        target_label: namespace
+      - action: replace
+        source_labels:
+        - __meta_kubernetes_pod_name
+        target_label: pod_name
+      - action: replace
+        source_labels:
+        - __meta_kubernetes_pod_container_name
+        target_label: container_name
+      - replacement: "/var/log/pods/*\$1/*.log"
+        separator: "/"
+        source_labels:
+        - __meta_kubernetes_pod_uid
+        - __meta_kubernetes_pod_container_name
+        target_label: __path__
+      - action: labelmap
+        regex: __meta_kubernetes_pod_label_(.+)
+    - job_name: journal
+      journal:
+        path: /var/log/journal
+        labels:
+          job: systemd-journal
+      pipeline_stages:
+      - labeldrop:
+        - filename
+        - stream
+      - pack:
+          labels:
+          - boot_id
+          - systemd_unit
+      - labelallow:
+          - host
+          - invoker
+      relabel_configs:
+      - action: labelmap
+        regex: __journal__(.+)
+    server:
+      http_listen_port: 3101
+    target_config:
+      sync_period: 10s
+---
+apiVersion: v1
+data:
+  cluster-name: build01
+kind: ConfigMap
+metadata:
+  name: cluster-name
+  namespace: loki
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: loki-promtail
+  namespace: loki
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: log-collector
+      app.kubernetes.io/instance: loki-promtail
+      app.kubernetes.io/name: promtail
+      app.kubernetes.io/part-of: loki
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: log-collector
+        app.kubernetes.io/instance: loki-promtail
+        app.kubernetes.io/name: promtail
+        app.kubernetes.io/part-of: loki
+        app.kubernetes.io/version: "2.2.1"
+    spec:
+      containers:
+      - args:
+        - --oidc.client-id=$(CLIENT_ID)
+        - --oidc.client-secret=$(CLIENT_SECRET)
+        - --oidc.issuer-url=https://sso.redhat.com/auth/realms/redhat-external
+        - --margin=10m
+        - --file=/tmp/shared/prod_bearer_token
+        name: prod-bearer-token
+        env:
+          - name: CLIENT_ID
+            valueFrom:
+              secretKeyRef:
+                name: promtail-prod-creds
+                key: client-id
+          - name: CLIENT_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: promtail-prod-creds
+                key: client-secret
+        volumeMounts:
+        - mountPath: "/tmp/shared"
+          name: shared-data
+        image: quay.io/observatorium/token-refresher
+      - command:
+        - sh
+        - -c
+        - |
+          promtail \
+            -client.external-labels=_id=$(cat /tmp/shared/cluster-id),host=$(HOSTNAME),invoker=$(cat /tmp/shared/cluster-invoker),cluster_name=$(CLUSTER_NAME) \
```

Comment:
Do we need `_id` and `invoker` labels here?

Reply:
No idea, I just copied it. Id might be nice I guess, `invoker` seems to be `hive` or `user` which doesn't seem particularly useful?